### PR TITLE
Goat projects prow run on code changes only

### PR DESCRIPTION
**Pull request by videlov**
**Description**

Changes proposed in this pull request:

- Run prow jobs for GOAT projects only on selected paths (exclude Docs/Perf tests etc.)

**Related issue(s)**

### CODE REVIEW

**Comment by Ressetkk** [comment]
I don't know if you are aware, but Prow also supports directive `skip_if_only_changed`. You can use that to simplify the regex overall. https://docs.prow.k8s.io/docs/jobs/#triggering-jobs-based-on-changes

**Comment by videlov** [comment]
> I don't know if you are aware, but Prow also supports directive `skip_if_only_changed`. You can use that to simplify the regex overall. https://docs.prow.k8s.io/docs/jobs/#triggering-jobs-based-on-changes

Ah cool, I was doing negation regex at first, this could be better with `skip_if_only_changed`. Thx will adapt it.

**Comment by videlov** [comment]
@Ressetkk `make jobs-definitions` did not render `skip_if_only_changed`, could you take a look?

File: templates/data/api-gateway-validation.yaml [hunk 1]
```diff
@@ -45,6 +48,7 @@ templates:
                     - linting
               - jobConfig:
                   name: pull-api-gateway-test
+                  skip_if_only_changed: '^docs/|^performance_tests/|\.md$|^(OWNERS|CODEOWNERS)$|^\.'
                   always_run: true
```
Comment (barchw) [inline templates/data/api-gateway-validation.yaml:52]: Shouldn't always run be removed?
```suggestion
```

**Review by barchw — CHANGES_REQUESTED** [review]

**Comment by Ressetkk** [comment]
> @Ressetkk `make jobs-definitions` did not render `skip_if_only_changed`, could you take a look?

Unfortunately, you will have to add this field into generic.tmpl. The same way as run_if_changed

**Review by Ressetkk — APPROVED** [review]
LGTM

**Pull request merged**